### ISSUE TITLE
Build error "Header 'Realm/RLMDictionary_Private.h' not found" in module.modulemap

### ISSUE THREAD

**Issue by nicolas255** [realm/realm-cocoa#7344]
## Goals

I am trying to use version of Realm in a SPM package.

## Expected Results

Xcode should resolve the dependencies successfully and then the package should build successfully.

## Actual Results

The dependency is resolved successfully but then the build fails with two errors :

![Screenshot 2021-07-17 at 21 04 43](https://user-images.githubusercontent.com/7323838/126047174-2594b0b4-a49b-4903-83bf-a45641e97406.png)

## Steps for others to Reproduce

1. Create a swift package with Xcode.
2. Add `realm-cocoa` as a dependency.
3. Start building.

## Code Sample

This is the `Package.swift` file :
```swift
let package = Package(
    name: "MyLibrary",
    platforms: [.iOS(.v14)],
    products: [
        .library(
            name: "MyLibrary",
            targets: ["MyLibrary"]
        ),
    ],
    dependencies: [
        .package(name: "Realm", url: "https://github.com/realm/realm-cocoa", .upToNextMajor(from: "10.10.0")),
    ],
    targets: [
        .target(
            name: "MyLibrary",
            dependencies: [
                .product(name: "Realm", package: "Realm"),
                .product(name: "RealmSwift", package: "Realm"),
            ]
        ),
        .testTarget(
            name: "MyLibraryTests",
            dependencies: [
                .target(name: "MyLibrary"),
                .product(name: "Realm", package: "Realm"),
                .product(name: "RealmSwift", package: "Realm"),
            ]
        ),
    ]
)
```

## Version of Realm and Tooling

Realm framework version:  
From `10.8.0` to `10.10.0`.  
Build succeeds with `10.7.7` or earlier.

Xcode version:  
`12.5.1`

iOS/OSX version:  
I tried targeting an iPhone running on iOS 14.6 and simulator running iOS 14.3.

Dependency manager + version:  
SPM, bundled with Xcode

**Comment by leemaguire** [comment]
Hi @nicolas255 I cannot reproduce your issue, have you cleaned your build and derived data?

**Comment by nicolas255** [comment]
Hi @leemaguire, I tested all of that but no luck there…

In the meantime, I also tried to upgrade from 10.7.4 to 10.10.0 in an existing project where packages are managed by Xcode (it's an iOS app project) and I have the same two issues. Then I tried to replicate them in a fresh iOS project and same issues again.

One thing that could be relevant (?) is that I am trying this on a case sensitive file system. At first I thought it could be that but I don't see any issue with the file names.

**Comment by nicolas255** [comment]
Another update: we managed to try on two additional macs. One of them (Intel) is also on a case sensitive FS and has the same issues. The other one (M1) is not on a case sensitive FS and it builds successfully… Both tried with Xcode 12.5.1.

I tried to see why that error could occur but I don't have any experience with packages that include anything other than Swift.

**Comment by schmidan** [comment]
I ran into the same issue with SPM and `10.10.0` downgrading to `10.7.7` helps.  
My system is _Case Sensitive_. I will report back when i get to test on an insensitive system.

for what its worth the apps realm is being upgraded from `3.21.0` :)

**Comment by nicolas255** [comment]
As soon as I have a little more time I will try to narrow down to the changes that cause this. Or at least to a specific commit…

**Comment by nicolas255** [comment]
So I found the issue, it's in the symlink there: https://github.com/realm/realm-cocoa/blob/267531b38c710f5a7f3352b34eb95ece53773093/include/Realm/RLMDictionary_Private.h which link to a `realm` directory instead of `Realm`.

I made a fork with a fix for it https://github.com/nicolas255/realm-cocoa/commit/5893c464fe2ffbcbed15c4bcb6574f999dcc3e15

And then it builds!

Could a contributor apply the same fix for the next patch/version ? I'm not sure about the [CLA](https://github.com/nicolas255/realm-cocoa/commit/5893c464fe2ffbcbed15c4bcb6574f999dcc3e15) thing and since it's not exactly a big change…

Thank you!

**Comment by nicolas255** [comment]
Could someone else look into this ? I'm kind of stuck with that question mark issue… 😟

**Comment by jsflax** [comment]
@nicolas255 we'll take a look at your PR today and try to get a release out later tonight.

**Comment by nicolas255** [comment]
@jsflax awesome !!

@schmidan could you try to upgrade to the revision of the PR (#7360) and see if you can build with it ??

Using :
```
.package(name: "Realm", url: "https://github.com/nicolas255/realm-cocoa", .branch("master")),
```
(or the equivalent using Xcode UI)

**Comment by leemaguire** [comment]
Closed via #7360
